### ISSUE TITLE
Modifying Service Templates: "htmlspecialchars() expects parameter 1 to be string, array given (View.php:127)"

### ISSUE THREAD

**Issue by tsvsj** [Icinga/icingaweb2-module-director#1502]
Hi,

I'm trying to edit a service template via Director. When clicking on "Edit" I get the PHP error, mentioned above.

This issue is reproducable and consists of two different steps:
1. Unvalidated parameter type in function "escape" in View.php
2. Transformation issue between data types of custom data fields
Therefore I will put 1 or 2 in front of the following descriptions.

## Current Behavior
1. Leads to the following PHP Error
```
htmlspecialchars() expects parameter 1 to be string, array given (View.php:127)

#0 [internal function]: Icinga\Application\ApplicationBootstrap->Icinga\Application\{closure}(2, 'htmlspecialchar...', '/usr/share/php/...', 127, Array)
#1 /usr/share/php/Icinga/Web/View.php(127): htmlspecialchars(Array, 58, 'UTF-8', true)
#2 /usr/share/icingaweb2/library/vendor/Zend/View/Helper/FormText.php(70): Icinga\Web\View->escape(Array)
#3 /usr/share/icingaweb2/library/vendor/Zend/View/Abstract.php(345): Zend_View_Helper_FormText->formText('var_disk_ignore...', Array, Array, NULL)
#4 /usr/share/php/Icinga/Web/View.php(289): Zend_View_Abstract->__call('formText', Array)
#5 /usr/share/icingaweb2/library/vendor/Zend/Form/Decorator/ViewHelper.php(254): Icinga\Web\View->__call('formText', Array)
#6 /usr/share/icingaweb2/library/vendor/Zend/Form/Element.php(2048): Zend_Form_Decorator_ViewHelper->render('')
#7 /usr/share/icingaweb2/library/vendor/Zend/Form/Decorator/FormElements.php(115): Zend_Form_Element->render()
#8 /usr/share/icingaweb2/library/vendor/Zend/Form/DisplayGroup.php(914): Zend_Form_Decorator_FormElements->render('')
#9 /usr/share/icingaweb2/library/vendor/Zend/Form/Decorator/FormElements.php(115): Zend_Form_DisplayGroup->render()
#10 /usr/share/icingaweb2/library/vendor/Zend/Form.php(2974): Zend_Form_Decorator_FormElements->render('')
#11 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Html/HtmlDocument.php(169): Zend_Form->render()
#12 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Html/BaseHtmlElement.php(112): dipl\Html\HtmlDocument->render()
#13 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Html/BaseHtmlElement.php(148): dipl\Html\BaseHtmlElement->renderContent()
#14 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Html/HtmlDocument.php(181): dipl\Html\BaseHtmlElement->render()
#15 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Zf1/SimpleViewRenderer.php(47): dipl\Html\HtmlDocument->__toString()
#16 /usr/share/icingaweb2/modules/director/library/vendor/ipl/Zf1/SimpleViewRenderer.php(66): dipl\Zf1\SimpleViewRenderer->render()
#17 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action/HelperBroker.php(272): dipl\Zf1\SimpleViewRenderer->postDispatch()
#18 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action.php(518): Zend_Controller_Action_HelperBroker->notifyPostDispatch()
#19 /usr/share/php/Icinga/Web/Controller/Dispatcher.php(76): Zend_Controller_Action->dispatch('editAction')
#20 /usr/share/icingaweb2/library/vendor/Zend/Controller/Front.php(937): Icinga\Web\Controller\Dispatcher->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#21 /usr/share/php/Icinga/Application/Web.php(409): Zend_Controller_Front->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#22 /usr/share/php/Icinga/Application/webrouter.php(104): Icinga\Application\Web->dispatch()
#23 /usr/share/icingaweb2/public/index.php(4): require_once('/usr/share/php/...')
#24 {main}
```

2. When the data type of a custom data field is changed, the data in linked objects are not correctly transformed/parsed to the destination data type. For example:
A linked data field of type array is filled with at least one element. After that the data type is changed to string. The elements of the data field are not transformed to type string (i.e. a comma-delimited list).

## Possible Solution
1. The public function escape within View.php does not sufficiently validate the type of $value. This should be corrected to support arrays.

2. Automatic transformation between data types (if possible) or prevent changing of the data type, as long as the object is linked.

## Steps to Reproduce (for bugs)
- Create custom data field of data type array
- Create or modify a service template and link the newly created data field.
- Put in at least one element in the array field of the service template.
- Change the data type of the data field to type string.
- Try to modify the service template, where the field is linked and contains at least one element

## Your Environment
Service Template
```
template Service "test" {
    import "generic-director-service"

    check_command = "hostalive"
    command_endpoint = host_name
    vars.test_array = [ "test_element" ]
}
```

* Director version (System - About): ??? master (Git Commit 05bc9e5bfc0e3068112919a606111cc6f89290d2)
* Icinga Web 2 version and modules (System - About): 2.5.3-1
* Icinga 2 version (`icinga2 --version`): 2.8.4-2
* Operating System and version: Raspbian/Debian 9.4 (Kernel 4.14.34-v7+)
* Webserver, PHP versions: 
Apache2, 2.4.25-3+deb9u4
PHP, 7.0.27-0+deb9u1, 7.2.4-1

**Comment by tsvsj** [comment]
I've fixed this issue by directly editing the database of the director (Table "icinga_service_var"): Changed "varvalue" from "[ "test_element" ]" to "test_element" (w/o quotes) and format from "json" to "string".

**Comment by Thomas-Gelf** [comment]
Will be fixed with #1200
